client/state/login/actions.js

```diff
@@ -350,3 +353,41 @@ export const sendSmsCode = () => ( dispatch, getState ) => {
 export const startPollAppPushAuth = () => ( { type: TWO_FACTOR_AUTHENTICATION_PUSH_POLL_START } );
 export const stopPollAppPushAuth = () => ( { type: TWO_FACTOR_AUTHENTICATION_PUSH_POLL_STOP } );
 export const formUpdate = () => ( { type: LOGIN_FORM_UPDATE } );
+
+/**
+ * Attempt to logout a user.
+ *
+ * @param  {String}    redirectTo         Url to redirect the user to upon successful logout
+ * @return {Function}                     Action thunk to trigger the logout process.
+ */
+export const logoutUser = ( redirectTo ) => dispatch => {
+	dispatch( {
+		type: LOGOUT_REQUEST,
+	} );
+
+	return request.post( 'https://wordpress.com/wp-login.php?action=logout-endpoint' )
```

Comment:
Mentioned on D6403, but this request should be sending the nonce embedded in the URL from `userUtils.getLogoutUrl()`.